File: app/src/main/java/com/trayis/simplimvvm/app/common/ConnectivityResource.java
package com.trayis.simplimvvm.app.common;

import android.content.Context;
import android.net.ConnectivityManager;
import android.os.Build;

import com.trayis.simplimvvm.utils.SimpliResource;

import androidx.annotation.RequiresApi;

public class ConnectivityResource implements SimpliResource {

    private final Context context;

    public ConnectivityResource(Context context) {
        this.context = context;
    }

    @RequiresApi(Build.VERSION_CODES.M)
    public boolean isConnected() {
        return context.getSystemService(ConnectivityManager.class).getActiveNetwork() != null;
    }

}

File: app/src/main/java/com/trayis/simplimvvm/app/common/ConnectivityResource.kt
package com.trayis.simplimvvm.app.common

import android.content.Context
import android.net.ConnectivityManager
import android.os.Build
import androidx.annotation.RequiresApi
import com.trayis.simplimvvm.utils.SimpliResource

/**
 * @author mudesai (Mukund Desai)
 * @created on 12/23/18
 */
class ConnectivityResource(val context: Context) : SimpliResource {

    @RequiresApi(Build.VERSION_CODES.M)
    fun isConnected(): Boolean {
        return context.getSystemService(ConnectivityManager::class.java).activeNetwork != null
    }

    companion object {
        fun getInstance(context: Context) = ConnectivityResource(context)
    }

}